integration-tests/src/main/java/io/druid/testing/clients/CoordinatorResourceTestClient.java

```diff
@@ -64,11 +65,35 @@ private String getCoordinatorURL()
     );
   }
 
+  private String getIntervalsURL(String dataSource)
+  {
+    return String.format("%sdatasources/%s/intervals", getCoordinatorURL(), dataSource);
+  }
+
   private String getLoadStatusURL()
   {
       return String.format("%s%s", getCoordinatorURL(), "loadstatus");
   }
 
+  // return a list of the segment dates for the specified datasource
+  public ArrayList<String> getSegmentDatesForDataSource(final String dataSource) throws Exception
+  {
+    ArrayList<String> segments = null;
+    try {
+      StatusResponseHolder response = makeRequest(HttpMethod.GET, getIntervalsURL(dataSource));
+
+      segments = jsonMapper.readValue(
+          response.getContent(), new TypeReference<ArrayList<String>>()
+          {
+          }
+      );
+    }
+    catch (Exception e) {
+      Throwables.propagate(e);
```

Comment:
I agree about "throw", but I think propagate() itself is supposed to throw an exception.  From 
http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/base/Throwables.html#propagate%28java.lang.Throwable%29:

 This method always throws an exception. The RuntimeException return type allows client code to signal to the compiler that statements after the call are unreachable.